lib/fathead/mdn_css/parse.pl

```diff
@@ -58,19 +104,221 @@
         say '';
+        say $code;
+        $code =~ s/\r?\n/\\n/g;
+    }
+    $hs->eof;
+    $description = build_abstract( $description, $code );
+
+    next unless $title && $link && $description;
+
+    #get external links
+    my $external_links;
+    my $div_external = $dom->at('div#toc');
+    if ($div_external) {
+        my $external_lis_collection = $div_external->find('li');
+
+        #link is used to make link absolute
+        $external_links =
+          make_external_links( $link, $external_lis_collection );
+    }
+    $external_links ||= '';
+
+    push @entries,
+      make_article( $title_clean, $description, $link, $external_links );
+
+    # Check for CSS Functions
+    # e.g. "not()"
+    # Replace "()" with "function" in title
+    # e.g. "not()" - > "not function"
+    if ( $title_clean =~ m/\(\)$/ ) {
+        say "Found Function: $title_clean";
+        my $temp = $title_clean;
+        $temp =~ s/\(\)$/ function/;
+        push @entries, make_redirect( $temp, $title );
+        $temp = $title_clean;
+        $temp =~ s/\(\)$//;
+        push @entries, make_redirect( $temp, $title );
+    }
+
+    write_to_file(@entries);
+}
+
+# PRIVATE FUNCTIONS
 
-        unless ( exists $titles_dedup{$title} ) {
-            $titles_dedup{$title} = 1;
-            my @data = (
-                $title, 'A', '', '', '', '', '', '', '', '', '', $description,
-                $link
-            );
-            say OUT join "\t", @data;
+sub clean_string {
+    my $input = shift;
+    my $clean = $input;
+
+    # Capture content inside parentheses
+    # Check if identical or not
+    # E.g. "::before (:before)"
+    my $paren_re = qr/\((.+)\)/;
+
+    $clean =~ s/[:<>@]//g;
+
+    if ( $clean ne $input ) {
+        say "Input: $input";
+        say "Cleaned: $clean";
+    }
+
+    if ( $clean =~ m/$paren_re/ ) {
+        say "Has parens!";
+        my $paren_text = trim($1);
+        say "Paren Text: $paren_text";
+        $clean =~ s/$paren_re//;
+        trim($clean);
+        say "Clean Now: $clean";
+        unless ( $clean eq $paren_text ) {
+            say "Paren Text is Different!";
+            $clean = "$clean $paren_text";
+        }
+        say "Clean Final: $clean";
+    }
+    return $clean;
+}
+
+sub build_abstract {
+    my ( $description, $code ) = @_;
+    say "NO DESCRIPTION!" if $description eq "";
+    my $out;
+    $out .= "<p>$description</p>" if $description;
+    $out .= "<br><pre><code>$code</code></pre>" if $code;
+    return $out;
+}
+
+sub make_article {
+    my ( $title, $description, $link, $external_links ) = @_;
+    say '';
+    say "TITLE: $title";
+    say "LINK: $link";
+    say "DESCRIPTION: $description" if $description;
+    say "EXTERNAL LINKS $external_links " if $external_links;
+    my @data = (
+        $title, 'A', '', '', '', '', '', '', $external_links || '',
+        '', '', $description, $link
+    );
+    return join "\t", @data;
+}
+
+sub make_external_links {
```

Comment:
I just realized something else though -- they have a `See Also` section -- this can definitely be used to generate `Related Topics`, similar to what you were doing here.

The Related Topics will **definitely** work here, creating an Infobox with links to articles (as long as we've index them)